src/main/runner/shims/browser/proxies/locator.ts

```diff
@@ -0,0 +1,143 @@
+import { Locator } from 'k6/browser'
+
+import { ActionLocator } from '@/main/runner/schema'
+
+import { ProxyOptions } from '../utils'
+
+export function locatorProxy(
+  target: Locator,
+  locator: ActionLocator
+): ProxyOptions<Locator> {
+  return {
+    target,
+    tracking: {
+      check(options) {
```

Comment:
These are the methods that take a call, e.g. `locator.check()`, and convert them into an event payload.